src/Orchard/Environment/ThreadProvider.cs

```diff
@@ -0,0 +1,9 @@
+using System.Threading;
+
+namespace Orchard.Environment {
+    public class ThreadProvider : IThreadProvider {
```

Comment:
I'd rather call it ThreadIdAccessor. It's not providing anything (other than access to the ID :-)).

Though I'm not sure it's too useful to keep track of the thread ID: in ASP.NET if the code is async multiple threads can handle the same request even.